indico/modules/events/registration/fields/simple.py

```diff
@@ -317,8 +318,12 @@ class CountryField(RegistrationFormFieldBase):
 
     @classmethod
     def unprocess_field_data(cls, versioned_data, unversioned_data):
+        def unaccent_cmp(x, y):
+            return locale.strcoll(x['caption'], y['caption'])
+
+        locale.setlocale(locale.LC_ALL, str(get_current_locale()))
```

Comment:
This looks rather problematic considering that it affects MUCH more than just sorting. `LC_COLLATE` should be enough. But again, ideally we can do this w/o modifying global state.

Reply:
So if I replace LC_ALL with LC_COLLATE, is it acceptable ?
Any other suggestion welcome.

Reply:
see the second half of my comment :) (`setlocale()` is always global)

Reply:
Babel unfortunately does not support locale-aware sorting. There is this [library](https://github.com/jtauber/pyuca) which doesn't seem to be actively maintained though. We'd like to avoid `setLocale` since it's not thread-safe, @vtran99 do you think the suggestion to strip accents and to convert to ascii would be sufficient?

Reply:
Code updated as suggested by Adrian, tested ok.

Reply:
I tested it with Czech and it's definitely better in some ways, but there are still some problems which we can't fix by just converting to ascii.

As an example, countries starting with `Č` are now mixed with countries starting with `C` as opposed to being all the way at the end of the list. This is definitely an improvement, but ideally `Č` should come after `C`. Another example is `Ch` being treated as `C` and not as its own letter which comes after `H`.

For me this is OK, but we should look into something that handles locales. Since there is [Intl.Collator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) I wonder if we could sort the list on the client side?

Reply:
For us, that behaviour is ok for users to easily find theirs countries, like: Égypte, États-Unis... as they would expect those names to be found between the "E.." in alphabetical order and not and the end.
So the same : "Č" to be sorted between "C" is correct from that point of view.
We can go with that if it's ok for Cern.